### PR TITLE
Rename OffChain to ObsCall and fix tx execution logs

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

- "OffChain" is not the best name for messages that execute view contract functions. "ObsCall" is more in line with "EthCall"
- We were logging smart contract execution errors as log.Error. They are not really errors, as they are expected, and were confusing the logs 

### What changes were made as part of this PR

- rename OffChain to ObsCall
- reduce the logging level of smart contract errors, and introduce "lazy" for some traces

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/evm/evm_facade.go
Comment (tudor-malene) [inline go/enclave/evm/evm_facade.go:1]: Logging changes only in this file

**Review by BedrockSquirrel — APPROVED** [review]
LGTM, minor comment about our logger treating pairs of arguments as key-value pairs. Plus looks like it needs a linting :)

File: go/enclave/l2chain/l2_chain.go [hunk 1]
```diff
@@ -240,16 +241,16 @@ func (oc *ObscuroChain) ExecuteOffChainTransactionAtBlock(apiArgs *gethapi.Trans
 		return nil, fmt.Errorf("unable to fetch head state batch. Cause: %w", err)
 	}
 
-	oc.logger.Trace(
-		fmt.Sprintf("!OffChain call: contractAddress=%s, from=%s, data=%s, batch=b_%d, state=%s",
+	oc.logger.Trace("Obs_Call:", "Successful result", gethlog.Lazy{Fn: func() string {
```
[the remark this answers is not in the document]
Comment (tudor-malene) [reply go/enclave/l2chain/l2_chain.go:244]: fixed. the lazy thing needs to be a value of something.
I split it up differently now

[the remark this answers is not in the document]
Comment (BedrockSquirrel) [reply go/enclave/l2chain/l2_chain.go:244]: Oh I see, yeah that's fine then. Could wrap the lazy output in braces or something to make it easier to read as
a value object but we've already spent too long talking about a Trace level log message so I'll be quiet now haha.

**Pull request merged**